service/src/main/java/com/derpgroup/derpwizard/manager/AbstractManager.java

```diff
@@ -64,6 +82,31 @@ public void handleRequest(@NonNull VoiceInput voiceInput, @NonNull SsmlDocumentB
     }
   }
 
+  private void appendToConversationHistory(@NonNull VoiceInput voiceInput) {
+    CommonMetadata metadata = voiceInput.getMetadata();
+    List<ConversationHistoryEntry> conversationHistory = metadata.getConversationHistory();
+    if(conversationHistory == null){
+      conversationHistory = new ArrayList<ConversationHistoryEntry>();
+    }
+    CommonMetadata metadataClone = null;
+    try {
+      ObjectMapper mapper = new ObjectMapper();
+      mapper.registerModule(mapperModule);
+      String metadataString = mapper.writeValueAsString(metadata);
+      metadataClone = mapper.readValue(metadataString, CommonMetadata.class);
```

Comment:
That's a perfectly reasonable question, and my original implementation was using convertValue, but unfortunately this results in runtime errors.  I think the `@JsonTypeInfo` and `@JsonSubType` annotations don't work with convertValue, for some reason.